Tests fail with `object has no attribute 'wrapped'` in propcache

### Please confirm the following

- [ ] I agree to follow the [aio-libs Code of Conduct][CoC]

[CoC]: ../../.github/blob/master/CODE_OF_CONDUCT.md

- [X] I have checked the [current issues][issue search] for duplicates.

[issue search]: ../search?q=is%3Aissue&type=issues

- [X] I understand this is open source software provided for free and that I might not receive a timely response.
- [X] I am positive I am **NOT** reporting a (potential) security
vulnerability, to the best of my knowledge. *(These must be shared by
submitting [this report form][vulnerability report form] instead, if
any hesitation exists.)*

[vulnerability report form]: ../security/advisories/new

- [ ] I am willing to submit a pull request with reporoducers as xfailing test cases or even entire fix. *(Assign this issue to me.)*

### Describe the bug

On 1.18.0 all tests have passed, on 1.18.3 some tests fail with:

```
    @benchmark
    def _run() -> None:
        for _ in range(100):
>           URL.raw_path_qs.wrapped(URL_WITH_PATH)
E           AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'

tests/test_url_benchmarks.py:714: AttributeError
```

Full log: https://gist.github.com/AMDmi3/a3a18e1ec0f9bdd839abcff7394a79c1

### To Reproduce

Run pytest

### Expected behavior

Test should pass

### Logs/tracebacks

```python-traceback
Full build log: https://gist.github.com/AMDmi3/a3a18e1ec0f9bdd839abcff7394a79c1
```


### Python Version

```console
3.11.11
```


### multidict Version

```console
6.1.0
```


### propcache Version

```console
0.2.0
```


### yarl Version

```console
1.18.3
```


### OS

FreeBSD 14.1

### Additional context

_No response_

Also see this on macOS 10.14.6 with python3.10 and yarl-1.18.3:
```
FAILED tests/test_url_benchmarks.py::test_empty_query_string_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
FAILED tests/test_url_benchmarks.py::test_empty_path_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
FAILED tests/test_url_benchmarks.py::test_empty_path_safe_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
FAILED tests/test_url_benchmarks.py::test_path_safe_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
FAILED tests/test_url_benchmarks.py::test_empty_raw_path_qs_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
FAILED tests/test_url_benchmarks.py::test_raw_path_qs_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
FAILED tests/test_url_benchmarks.py::test_raw_path_qs_with_query_uncached - AttributeError: 'propcache._helpers_c.under_cached_property' object has no attribute 'wrapped'
========================================================================================= 7 failed, 1459 passed, 4 xfailed in 7.97s ==========================================================================================
```

git bisect says 7585a9694990f4b60e43f49f0a5f2f0ab8ef24f6 is the culprit:
```
7585a9694990f4b60e43f49f0a5f2f0ab8ef24f6 is the first bad commit
commit 7585a9694990f4b60e43f49f0a5f2f0ab8ef24f6
Author: J. Nick Koston <nick@koston.org>
Date:   Sun Dec 1 12:57:50 2024 -0600

    Add benchmarks that bypass the cached property decorators (#1445)

 tests/test_url_benchmarks.py | 117 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 117 insertions(+)

```

--- a/CHANGES/1479.contrib.rst
+++ b/CHANGES/1479.contrib.rst
@@ -1,0 +1,4 @@
+Increased minimum `propcache`_ version to 0.2.1 to fix failing tests -- by :user:`bdraco`.
+
+.. _`propcache`:
+   https://github.com/aio-libs/propcache

--- a/setup.cfg
+++ b/setup.cfg
@@ -67,7 +67,7 @@ include_package_data = True
 install_requires =
   idna >= 2.0
   multidict >= 4.0
-  propcache >= 0.2.0
+  propcache >= 0.2.1
 
 [options.package_data]
 # Ref:

